src/plugins/FixRegion.cpp

```diff
@@ -46,16 +46,16 @@ void FixRegion::init(ParticleContainer* particleContainer, DomainDecompBase* dom
 
 	_molCount = 0;
 
+	std::array<double, 3> min{_xMin, _yMin, _zMin};
+	std::array<double, 3> max{_xMax, _yMax, _zMax};
 	// ITERATE OVER PARTICLES
-	for (auto tm = particleContainer->iterator(ParticleIterator::ONLY_INNER_AND_BOUNDARY); tm.isValid(); ++tm) {
-		if (tm->r(0) >= _xMin && tm->r(0) <= _xMax && tm->r(1) >= _yMin && tm->r(1) <= _yMax && tm->r(2) >= _zMin &&
-			tm->r(2) <= _zMax) {
-			for (unsigned i = 0; i < 3; i++) {
-				tm->setv(i, 0.0);
-				tm->setF(i, 0.0);
-			}
-			_molCount++;
+	for (auto tm = particleContainer->regionIterator(min.data(), max.data(), ParticleIterator::ONLY_INNER_AND_BOUNDARY);
```

Comment:
OK, so I did a bunch of tests and this part here didn't work. What happens is that the regionIterator basically doesn't do anything, i.e. no particles are examined and then the simulation crashes when attempting to do the global communication step for the variable _molCount. We then tried to rework this part, although I am not sure I defined the borders of the region correctly since it still does not work: 

```
double regionLowCorner[3], regionHighCorner[3];
	
	if(particleContainer->getBoundingBoxMin(0)>=_xMin && particleContainer->getBoundingBoxMin(1)>=_yMin && particleContainer->getBoundingBoxMin(2)>=_zMin &&
		particleContainer->getBoundingBoxMax(0)<=_xMax && particleContainer->getBoundingBoxMax(1)<=_yMax && particleContainer->getBoundingBoxMax(2)<=_zMax) {
			for(unsigned d = 0; d < 3; d++){
				regionLowCorner[d] = particleContainer->getBoundingBoxMin(d);
				regionHighCorner[d] = particleContainer->getBoundingBoxMax(d);
			}
		} 
#if defined (_OPENMP)
#pragma omp parallel shared(regionLowCorner, regionHighCorner)
#endif
	
	// ITERATE OVER PARTICLES
	for (auto temporaryMolecule = particleContainer->regionIterator(regionLowCorner, regionHighCorner, ParticleIterator::ALL_CELLS);
		 temporaryMolecule.isValid(); ++temporaryMolecule) {
		for (unsigned i = 0; i < 3; i++) {
			temporaryMolecule->setv(i, 0.0);
			temporaryMolecule->setF(i, 0.0);
		}
		global_log->info() << "A" << std::endl;
		temporaryMolecule->clearFM();
		_molCount++;
	}
```

Could you have a look @SteffenSeckler ?

Reply:
see above. you also don't want to use ParticleIterator::ALL_CELLS.